File: core/JEj789/JEj789.py
from typing import List


def minCost(costs: List[List[int]]) -> int:
    first, second, third = 0, 0, 0
    for item1, item2, item3 in costs:
        item1 += min(second, third)
        item2 += min(first, third)
        item3 += min(first, second)
        first, second, third = item1, item2, item3

    return min(first, min(second, third))

File: core/JEj789/JEj789.java
package com.dbc.code;

public class JEj789 {
    public int minCost(int[][] costs) {
        int first = 0, second = 0, third = 0;
        for (int[] cost : costs) {
            int item1 = cost[0] + Math.min(second, third);
            int item2 = cost[1] + Math.min(first, third);
            int item3 = cost[2] + Math.min(first, second);
            first = item1;
            second = item2;
            third = item3;
        }
        return Math.min(first, Math.min(second, third));
    }
}
